### DISCUSSION TITLE
[HELP] Insufficient instructions for installing Libadwaita

### DISCUSSION THREAD

**Discussion by natewind** [gtk-rs/gtk4-rs#1885] [General]
First of all, it’s not even mentioned [here](https://gtk-rs.org/gtk4-rs/stable/latest/book/libadwaita.html) that you need to add `PKG_CONFIG_PATH=/usr/lib/x86_64-linux-gnu/pkgconfig` to `.bashrc`  (at least on Ubuntu). Installing `libadwaita-1-dev` doesn’t do it itself for some unfathomable reason.

Second, it doesn’t even help, the same error still happens:

```
error: failed to run custom build command for `libadwaita-sys v0.7.0`

Caused by:
  process didn't exit successfully: `/<...>/target/debug/build/libadwaita-sys-53426ef3de26df35/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=/home/natewind/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libadwaita-sys-0.7.0/Cargo.toml
  cargo:rerun-if-env-changed=LIBADWAITA_1_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=
  pkg-config exited with status code 1
  > PKG_CONFIG_PATH=/usr/lib/x86_64-linux-gnu/pkgconfig PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 pkg-config --libs --cflags libadwaita-1 libadwaita-1 >= 1.5

  The system library `libadwaita-1` required by crate `libadwaita-sys` was not found.
  The file `libadwaita-1.pc` needs to be installed and the PKG_CONFIG_PATH environment variable must contain its parent directory.
  PKG_CONFIG_PATH contains the following:
      - /usr/lib/x86_64-linux-gnu/pkgconfig

  HINT: you may need to install a package such as libadwaita-1, libadwaita-1-dev or libadwaita-1-devel.
```

Even though the file is there:

```bash
~$ ls $PKG_CONFIG_PATH | grep libadwaita-1.pc
libadwaita-1.pc
```

**Comment by natewind** [comment]
**Upd:** Fixed by setting `features = ["v1_1"]`, the same as `pkg-config --modversion libadwaita-1`, but this whole process should definitely be explained in the book.

**Comment by bilelmoussaoui** [comment]
>PKG_CONFIG_PATH=/usr/lib/x86_64-linux-gnu/pkgconfig

That is your machine being broken. The same with the rest of your issue

**Comment by natewind** [comment]
> That is your machine being broken. The same with the rest of your issue

How is it broken and why would it be? I don’t think I did anything with `PKG_CONFIG_PATH`.

**Comment by bilelmoussaoui** [comment]
I am not an Ubuntu expert nor a help forum. But yes, installing the devel version should just work, like it works for hundreds of people using this crate already.

For the other issue, the book needs libadwaita v1.5, as described by the feature used. Your version is too old, that is why switching to v1_1 "solves" the problem.

**Comment by natewind** [comment]
> But yes, installing the devel version should just work, like it works for hundreds of people using this crate already.

Hmm, I might try to reproduce it later on a live USB with fresh Ubuntu.

> Your version is too old, that is why switching to v1_1 "solves" the problem.

The book recommends `sudo apt install libadwaita-1-dev`, and this it the [version](https://packages.ubuntu.com/jammy/libadwaita-1-dev) it installs on Ubuntu 22.04, which is currently the secend newest Ubuntu LTS and supported until 2032. If a third-party PPA with a newer version is required, it should be mentioned in the book. Or if the version should be changed according to the one your distro provides, it should also be mentioned. It’s weird to expect everybody to run the absolute newest version of their distro.

**Comment by bilelmoussaoui** [comment]
It is also weird to expect the book to mention every single LTS distro that ever exists on the planet and keep the information up to date.

The book doesn't have any active maintainers/authors to keep it up to date or fix any issues it might contain.

**Comment by natewind** [comment]
Fair enough, but at least it would be nice to mention that you might not have the latest version and should check it if it doesn’t compile. And that the version goes to `features = ...` specifically.

**Comment by bilelmoussaoui** [comment]
Sure thing, if you send a patch I will review and merge it, otherwise the situation will probably not change ;)

**Comment by natewind** [comment]
Ok, I’ll try to find time at some point.
